R/pkg/R/SQLContext.R

```diff
@@ -90,19 +96,31 @@ createDataFrame <- function(sqlContext, data, schema = NULL, samplingRatio = 1.0
       if (is.null(schema)) {
         schema <- names(data)
       }
-      n <- nrow(data)
-      m <- ncol(data)
-      # get rid of factor type
-      dropFactor <- function(x) {
+
+      # types that should be wrapped before
+      # mapply to avoid flattening
+      requires_wrapping <- c("list")
+
+      # get rid of factor type and adjust for lists
+      cleanCols <- function(x) {
         if (is.factor(x)) {
           as.character(x)
+        } else if(class(x) %in% requires_wrapping) {
```

Comment:
ah, I think you were trying to emulating a behavior of a bug when creating a DataFrame from a data.frame having any column of list type. Actually, this bug is https://issues.apache.org/jira/browse/SPARK-11283 reported by you:)

So this PR not only is a performance improvement, but also fixes SPARK-11283.

please remove code piece here:)